dsa-solutions/lc-solutions/1900-1999/1902-depth-of-bst-given-insertion-order.md

```diff
@@ -0,0 +1,285 @@
+---
+id: depth-of-bst-given-insertion-order
+title: Depth of BST Given Insertion Order
+sidebar_label: 1902-depth-of-bst-given-insertion-order
+tags:
+- Binary Search Tree
+- Tree
+- Binary Tree
+- Ordered Set
+description: "This document provides solutions for depth-of-bst-given-insertion-order."
+---
+
+## Problem
+
+You are given an array order which is a permutation of integers from 1 to n. This array represents the sequence in which nodes are inserted into a binary search tree (BST). The properties of a BST are such that:
+
+  1- The left subtree of a node contains only nodes with keys less than
+     the node's key.
+  2- The right subtree of a node contains only nodes with keys greater
+      than the node's key.
+  3- Both the left and right subtrees must also be binary search trees.
+
+The first element of order is the root of the BST. All subsequent elements are inserted one by one, at the correct position to maintain the BST property. You need to find the depth of this binary search tree, which is the number of nodes along the longest path from the root node down to the farthest leaf node.
+
+### Examples
+
+**Example 1:**
+
+Input: order = [2,1,4,3]
+Output: 3
+Explanation: The binary search tree has a depth of 3 with path 2->3->4.
+
+**Example 2:**
+
+Input: order = [2,1,3,4]
+Output: 3
+Explanation: The binary search tree has a depth of 3 with path 2->3->4.
+
+**Example 3:**
+
+Input: order = [1,2,3,4]
+Output: 4
+Explanation: The binary search tree has a depth of 4 with path 1->2->3->4.
```

Comment:
Please remember the following text: 

Replace `1->2->3->4` with ``` `1->2->3->4` ```.

Reply:
made all the changes